pkg/challenge/challenge_test.go

```diff
@@ -0,0 +1,36 @@
+package challenge
```

Comment:
I put some failure stuff in the cross tests, I kind of view the other tests as "just make sure it doesn't blow up" =) ... because otherwise we get a lot of duplication.

Reply:
Fair! Let's say good enough for now. We can always add more